File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.cs
using System;

public class Solution {
    public int solution(string[] babbling) {
        int answer = 0;
        string[] words = {"aya", "ye", "woo", "ma"};
        string temp = "";
        foreach(var s in babbling)
        {
            string cur = s;
            bool isOK = true;
            foreach(var word in words)
            {
                if(cur.Contains(word + word)) isOK = false;
                cur = cur.Replace(word, " ");
            }
            if(!isOK) continue;
            if(cur.Trim().Length == 0) answer++;
        }
        return answer;
    }
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<string> babbling) {
    int answer = 0;
    string words[] = {"aya", "ye", "woo", "ma"};

    for(string s : babbling)
    {
        int i = 0;
        int prev = -1;
        bool ok = true;
        
        while(i < s.size())
        {
            bool matched = false;
            for(int j = 0; j < 4; j++)
            {
                string word = words[j];
                if(s.substr(i, word.size()) == word)
                {
                    if(prev == j)
                    {
                        ok = false;
                        matched = true;
                        break;
                    }
                    prev = j;
                    i += word.size();
                    matched = true;
                    break;
                }
            }
            if(!matched)
            {
                ok = false;
                break;
            }
            if(!ok) break;                
        }
        if(ok) answer++;
    }
    return answer;
}